### ISSUE TITLE
AutoEncoder building for automatic feature extraction

### ISSUE THREAD

**Issue by gcoter** [joeddav/devol#25]
Hello,

I would like to share an idea I recently had.

Currently, I have the feeling that there are two kinds of auto-ml tools: those which build neural networks (and yours looks very promising) and those which tune and combine classifiers or regressors (auto-sklearn and TPOT for instance).

I think we would benefit from using both in some cases. For example, suppose I want to classify images. I can't use auto-sklearn or TPOT directly on images. However, I could:

1. Use DEvol to train a multilayer autoencoder which minimizes the reconstruction error
2. Use this trained autoencoder as a sklearn transformer to retrieve the extracted features (in the middle layer)
3. Provide those features to TPOT and/or auto-sklearn

The final result is a sklearn pipeline which has been built automatically from the raw dataset. In a sense, it is similar to the idea @ClimbsRocks discussed in #15. But instead of using only a gradient boosted model with the extracted features, we just let auto-sklearn and/or TPOT find the best combination of models.

What do you think about this idea? How much work would be necessary to allow DEvol to build autoencoder?
